R/addins.R

```diff
@@ -0,0 +1,111 @@
+#' Style the active file
+#'
+#' Helper function for RStudio Addin.
+style_active_file <- function() {
+  file <- rstudioapi::getActiveDocumentContext()$path
+  style_file(file, style = tidyverse_style)
+}
+
+
+#' Style the highlighted region
+#'
+#' Helper function for RStudio Addin. This function is complicated because of
+#' one thing: You can highlight also just parts of lines.
+#' @importFrom rlang seq2
+style_active_region <- function() {
+  context <- find_active_context()
+  all_text <- utf8::read_lines_enc(context$path)
+  styled_expr <- style_region(all_text, context)
+  neighbourhood <- extract_neighbourhood(all_text, context)
+
+  styled_lines <- complete_styled_expr(context, styled_expr, neighbourhood)
+
+  merged_text <- append(neighbourhood$body, styled_lines, context$start[1] - 1)
+  utf8::write_lines_enc(merged_text, context$path)
+}
+
+#' Helper to extract and preprocess relevant attributes from
+#' [rstudioapi::getActiveDocumentContext()].
+#'
+find_active_context <- function() {
+  context <- rstudioapi::getActiveDocumentContext()
+  path <- context$path
+  start <- context$selection[[1]]$range$start
+  end <- context$selection[[1]]$range$end
+  if (all(start == end)) return()
```

Comment:
... and then return a list with only one component here?

Reply:
I think this should rather be `stop("No region selected")` instead of `return()`.

Reply:
Even if we want to call this for getting the active file?

Reply:
No then not. But currently, we don't use find_active_context() for styling files. Should we? If yes, then the error message should not be implemented within the function find_active_context, I agree.

Reply:
It's currently only used for styling regions within files.

Reply:
So you mean we add an argument whole_file to to find_active_context() so it can be used for styling files as well as for highlighted regions?

Reply:
Ok, I adapted it so it is used in both cases.